js/components/debug.tsx

```diff
@@ -81,171 +55,239 @@ class DebugViewer extends React.Component<IComponentProps, IDebugViewerState> {
+                    </table>
+                </div>
+            </li>
+        );
     }
 
     renderLogEntries(entries) {
-
-        const isExpanded = this.state.toggle['executionlog'];
-
-        return <div className={'debug-root'} key={'debug-root'}>
-            <div
-                className={'debug-root-toggle'}
-                id={'executionlog'}
-                onClick={this.toggleHeader}>
-
-                <span className={
-                    'glyphicon glyphicon-triangle-' + ((isExpanded) ? 'bottom' : 'right')
-                } />
-                <h5 className={'debug-title'}>Execution Log</h5>
-                <span className={'label label-info'}>{entries.length}</span>
-            </div>
-            <div className={isExpanded ? null : 'hidden'}>
-                <table className={'table table-striped'}>
-                    <tr>
-                        <th>Timestamp</th><th>Message</th><th>Data</th>
-                    </tr>
-                    {
-                        manywho.utils.convertToArray(entries).map((entry) => {
-
-                            const timeStamp = new Date(entry.timestamp);
-
-                            return <tr key={entry.timestamp}>
-                                <td>{timeStamp.toLocaleString()}</td>
-                                <td>{entry.message}</td>
-                                { /* TODO: display data */}
-                            </tr>;
-
-                        })
-                    }
-                </table>
+        const isExpanded = this.state.toggle.executionlog;
+
+        return (
+            <div className="debug-root" key="debug-root">
+                <div
+                    className="debug-root-toggle"
+                    id="executionlog"
+                    onClick={this.toggleHeader}
+                    role="presentation"
+                >
+
+                    <span className={`glyphicon glyphicon-triangle-${((isExpanded) ? 'bottom' : 'right')}`} />
+                    <h5 className="debug-title">Execution Log</h5>
+                    <span className="label label-info">{entries.length}</span>
+                </div>
+                <div className={isExpanded ? null : 'hidden'}>
+                    <table className="table table-striped">
+                        <tr>
+                            <th>Timestamp</th>
+                            <th>Message</th>
+                            <th>Data</th>
+                        </tr>
+                        {
+                            manywho.utils.convertToArray(entries).map((entry) => {
+
+                                const timeStamp = new Date(entry.timestamp);
+
+                                return (
+                                    <tr key={entry.timestamp}>
+                                        <td>{timeStamp.toLocaleString()}</td>
+                                        <td>{entry.message}</td>
+                                        { /* TODO: display data */}
```

Comment:
Should this be done? Or can it be removed?

Reply:
There is a data key in the execution log objects returned from the engine, but from what I've seen so far, they always appear to be null. So may as well keep the todo, as this may be something that gets implemented/is required in the future.